### PR TITLE
Added Pagination 

### PR DESCRIPTION
**Pull request by aaronNgu**
Watch it live [here](https://coursehububc.herokuapp.com/homepage)

This one is quite big of a ticket. Essentially I added pagination to manage more courses.

In summary, what I did here are the following: 
1. Frontend for Pagination.
2. Redux to make Pagination Component a controlled component 
3. Endpoints to manage pagination. This involves adding pagenumber as a parameter of the url.(/courses?page=1)
4. Change data structure of the get request. Initially, `/courses` returns an array of courses. Now it returns 
`{pageCount: <number of pages to display all the documents>, 
   data: <array of courses>}`. 
5. Amend the api call to the backend. added parameter to the url.

I will add more comments in the code to differentiate the parts.

### CODE REVIEW

File: routes/courses.js [hunk 1]
```diff
@@ -5,13 +5,24 @@ const Course = require('../models/course')
...
       .populate('Course')
         .exec()
         .then(docs =>{
-          console.log(docs);
-          res.status(200).json(docs);
+          Course.countDocuments()
+            .then(count => {
+              package = { 
+                pageCount : Math.floor(count/10) + 1,
+                data: docs};
```
Comment (aaronNgu) [inline routes/courses.js:20]: Forming response `{pageCount: <integer>, data: [<courses1>, <courses2>]}`

File: routes/courses.js [hunk 2]
```diff
@@ -5,13 +5,24 @@ const Course = require('../models/course')
 
 /* GET Courses listing. */
 router.get('/', function (req, res, next) {
+
+  const pagenumber = typeof req.query.page === 'undefined' ? 0 : (req.query.page - 1 );
+
   res.setHeader('Content-Type', 'application/json');
-  Course.find()
+  Course.find({},{}, {skip: 10 * pagenumber, limit:10})
```
Comment (aaronNgu) [inline routes/courses.js:12]: Returns at most 10 courses `limit:10` while skipping some number of courses depending on the page `skip: 10 * pagenumber`

File: react_app/src/reducers/index.js [hunk 3]
```diff
@@ -55,7 +54,9 @@ const allReducers = combineReducers({
     toggleAddReviewBox: addReviewBox,
     addReviewRating: addReviewRating,
     addReviewReview: addReviewReview,
-    reviewList: reviewReducer
+    reviewList: reviewReducer,
+    countPage: countPage, 
+    currentPage: currentPage,
```
Comment (aaronNgu) [inline react_app/src/reducers/index.js:59]: This determines what page we're viewing.

File: react_app/src/reducers/index.js [hunk 4]
```diff
@@ -55,7 +54,9 @@ const allReducers = combineReducers({
     toggleAddReviewBox: addReviewBox,
     addReviewRating: addReviewRating,
     addReviewReview: addReviewReview,
-    reviewList: reviewReducer
+    reviewList: reviewReducer,
+    countPage: countPage,
```
Comment (aaronNgu) [inline react_app/src/reducers/index.js:58]: determines how many pages there are. In other words, how many pages are needed to display all the courses.

File: react_app/src/actions/index.js [hunk 5]
```diff
@@ -6,13 +6,15 @@ export const fetched_courses = courses => {
...
     return function (dispatch, getState) {
-        return fetch(`/courses`)
+        const url = `/courses?page=` + currentPage;
+        return fetch(url)
             .then(
                 data => data.json())
             .then(data => {
-                    dispatch(fetched_courses(data))
+                    dispatch(change_page_count(data['pageCount'])) 
+                    dispatch(fetched_courses(data['data']))
```
Comment (aaronNgu) [inline react_app/src/actions/index.js:17]: With the changes from the backend, `data` now looks like `{pageCount: <integer>, data: [<courses1>, <courses2>]}`

**Review by illeanz — APPROVED** [review]
Thank you! :)

**Review by wwyc — APPROVED** [review]

**Pull request merged**